File: task_2.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cctype> // для isdigit() и isalpha()

using namespace std;

void separateLettersAndDigits(const vector<string>& inputList) {
    vector<string> letters;
    vector<string> digits;

    for (const auto& s : inputList) {
        if (s.empty()) continue; // Пропускаем пустые строки

        char c = s[0]; // Берем первый символ строки

        if (isalpha(c)) {
            letters.push_back(s);
        } else if (isdigit(c)) {
            digits.push_back(s);
        }
    }

    // Выводим результаты
    cout << "Буквы: ";
    for (const auto& letter : letters) {
        cout << letter << " ";
    }
    cout << endl;

    cout << "Цифры: ";
    for (const auto& digit : digits) {
        cout << digit << " ";
    }
    cout << endl;
}

int main() {
    // Пример использования
    vector<string> input = {"a", "1", "b", "2", "c", "3"};
    separateLettersAndDigits(input);

    return 0;
}

File: task_2.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        // Пример вызова функции с тестовыми данными
        List<String> input = List.of("a", "1", "b", "2", "c", "3");
        separateLettersAndDigits(input);
    }

    public static void separateLettersAndDigits(List<String> inputList) {
        // Создаем два списка для букв и цифр
        List<String> letters = new ArrayList<>();
        List<String> digits = new ArrayList<>();

        // Проходим по всем элементам входного списка
        for (String s : inputList) {
            // Извлекаем первый символ строки (т.к. строка состоит из 1 символа)
            char c = s.charAt(0);

            if (Character.isLetter(c)) {
                letters.add(s); // Добавляем в список букв
            } else if (Character.isDigit(c)) {
                digits.add(s); // Добавляем в список цифр
            }
        }

        // Выводим результаты на экран
        System.out.println("Буквы: " + letters);
        System.out.println("Цифры: " + digits);
    }
}